### PR TITLE
docs: Add cookbook recipe to copy file to Engine runtime container

### PR DESCRIPTION
**Pull request by vikram-dagger**
Based on discussion in https://discord.com/channels/707636530424053791/1300820432957476916/1300858765297582211

### CODE REVIEW

File: docs/current_docs/cookbook/cookbook.mdx [hunk 1]
```diff
@@ -192,6 +192,38 @@ The following Dagger Function accepts a `Directory` argument, which could refere
       directory --path=/src \
       entries
     ```
+### Copy a file to the Dagger Engine runtime container for custom processing
+
+The following Dagger Function accepts a `File` argument and copies the specified file to the Dagger Engine module runtime container. This makes it possible to add one or more files to the runtime container and manipulate them using custom logic.
```
Comment (marcosnils) [inline docs/current_docs/cookbook/cookbook.mdx:197]:
```suggestion
The following Dagger Function accepts a `File` argument and copies the specified file to the Dagger Module runtime container. This makes it possible to add one or more files to the runtime container and manipulate them using custom logic.
```

I think `Engine` in this sentence brings some confusion.

File: docs/current_docs/cookbook/snippets/copy-dir-runtime/go/main.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"context"
+	"dagger/my-module/internal/dagger"
+	"fmt"
+	"os"
+)
+
+type MyModule struct{}
+
+// Copy a file to the Dagger Engine runtime container for custom processing
```
Comment (marcosnils) [inline docs/current_docs/cookbook/snippets/copy-dir-runtime/go/main.go:12]:
```suggestion
// Copy a file to the Dagger  Module Go runtime container for custom processing
```

**Review by marcosnils — COMMENTED** [review]
LGTM! left a few docs nits.

**Review by marcosnils — APPROVED** [review]

File: docs/current_docs/cookbook/snippets/copy-dir-runtime/python/main.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+@object_type
+class MyModule:
+    @function
+    async def copy_file(self, source: dagger.File):
+        """Copy a file to the Dagger module runtime container for custom processing"""
+        await source.export("foo.txt")
+        # your custom logic here
+        # for example, read and print the file in the Dagger Engine container
+        print(open("foo.txt").read())
```
Comment (helderco) [inline docs/current_docs/cookbook/snippets/copy-dir-runtime/python/main.py:13]: The standard way to do this in Python is with [pathlib](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text):
```diff
- open("foo.txt").read()
+ pathlib.Path("foo.txt").read_text()
```

However, that's a sync (blocking) operation, so in Dagger you should use [anyio](https://anyio.readthedocs.io/en/stable/fileio.html) for asynchronous I/O operations like disk or network access:
```suggestion
        print(await anyio.Path("foo.txt").read_text())
```

Comment (helderco) [reply docs/current_docs/cookbook/snippets/copy-dir-runtime/python/main.py:13]: Don't forget:
```python
import anyio
```

**Pull request merged**